clang/test/CodeGenSYCL/intel-sycl-ptr-annotation-builtin.cpp

```diff
@@ -0,0 +1,140 @@
+  // CHECK-DAG: [[PTR4:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[A2]]{{.*}}[[ARG2]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR4]], ptr addrspace(4) %x
+  x = __builtin_intel_sycl_ptr_annotation(A, "testB", 127);
+
+  // CHECK-DAG: [[B2:%[0-9]+]] = load ptr addrspace(4), ptr addrspace(4) [[Baddr]]
+  // CHECK-DAG: [[PTR5:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[B2]]{{.*}}[[ARG2]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR5]], ptr addrspace(4) %y
+  y = __builtin_intel_sycl_ptr_annotation(B, "testB", 127);
+
+  // CHECK-DAG: [[C2:%[0-9]+]] = load ptr addrspace(4), ptr addrspace(4) [[Caddr]]
+  // CHECK-DAG: [[PTR6:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[C2]]{{.*}}[[ARG2]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR6]], ptr addrspace(4) %z
+  z = __builtin_intel_sycl_ptr_annotation(C, "testB", 127);
+
+  // CHECK-DAG: [[D:%[0-9]+]] = load ptr addrspace(4), ptr addrspace(4) [[Daddr]]
+  // CHECK-DAG: [[PTR7:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[D]]{{.*}}[[ARG2]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR7]], ptr addrspace(4) %z
+  z = __builtin_intel_sycl_ptr_annotation(&D, "testB", 127);
+
+  // CHECK-DAG: [[A3:%[0-9]+]] = load ptr addrspace(4), ptr addrspace(4) [[Aaddr]]
+  // CHECK-DAG: [[PTR9:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[A3]]{{.*}}[[ARG3]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR9]], ptr addrspace(4) %x
+  x = __builtin_intel_sycl_ptr_annotation(A, "testC", "testC", "testC", 0, 0, 0);
+
+  // CHECK-DAG: [[A4:%[0-9]+]] = load ptr addrspace(4), ptr addrspace(4) [[Aaddr]]
+  // CHECK-DAG: [[PTR10:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[A4]]{{.*}}[[ARG4]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR10]], ptr addrspace(4) %x
+  x = __builtin_intel_sycl_ptr_annotation(A, "testD", "testD", "testD", 0, 0, 0);
+
+  // CHECK-DAG: [[B3:%[0-9]+]] = load ptr addrspace(4), ptr addrspace(4) [[Baddr]]
+  // CHECK-DAG: [[PTR11:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[B3]]{{.*}}[[ARG5]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR11]], ptr addrspace(4) %y
+  y = __builtin_intel_sycl_ptr_annotation(B, "testE", "testE", 0, 0);
+
+  constexpr int TestVal1 = 7;
+  constexpr int TestVal2 = 8;
+
+  // CHECK-DAG: [[D1:%[0-9]+]] = load ptr addrspace(4), ptr addrspace(4) [[Daddr]]
+  // CHECK-DAG: [[PTR12:%[0-9]+]] = call ptr addrspace(4) @llvm.ptr.annotation{{.*}}[[D1]]{{.*}}[[ARG6]]{{.*}}
+  // CHECK-DAG: store ptr addrspace(4) [[PTR12]], ptr addrspace(4) %z
+  z = __builtin_intel_sycl_ptr_annotation(&D, "testF", "testF", "testF", "testF", 0, 0, TestVal1, TestVal2);
+}
+
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(const Func &kernelFunc) {
+  kernelFunc();
+
+  int *a;
+  annotation_with_load(a);
+  annotation_with_store(a);
+}
+
+int main() {
+  kernel_single_task<class fake_kernel>([]() {
```

Comment:
Please align all clang tests with https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#dpc-clang-fe-tests .

Reply:
Sorry we might not have enough time to do this before CC

Reply:
Is it? I'm not seeing where `__SYCL_DEVICE_ONLY__` defined in this file.

Reply:
```
// RUN: %clang_cc1 -fsycl-is-device -sycl-std=2020 -verify -pedantic -fsyntax-only -internal-isystem %S/Inputs %s
// RUN: %clang_cc1 -verify -pedantic -fsyntax-only -x c++ %s

#include "sycl.hpp"

int main() {
#ifdef __SYCL_DEVICE_ONLY__
#error "should be not raised"
#endif
}
```
this error will be hit. So it won't be able to test the error msg